stages/org.osbuild.systemd.unit

```diff
@@ -51,6 +62,11 @@ SCHEMA = r"""
         }
       }
     }
+  },
+  "userunit": {
```

Comment:
Torn between `userunit`, `user-unit`, or making it an enum.  Is it possible we might need a third value at some point?

Reply:
`user-unit` makes more sense to me. I doubt we need a third one systemd has two level : system and user, system is the most used thus it defaults it to false.

Reply:
Looking at the `edit` section of the systemctl manpage [1], there are actually three options, `--system`, `--user`, and `--global`.  And in fact, the type of drop-in we want to create with this change is the equivalent of `--global` rather than `--user`.

For example, running:
```
systemctl edit --user grub-boot-success.timer
```
creates/edits the file `/root/.config/systemd/user/grub-boot-success.timer.d/override.conf`
whereas running:
```
systemctl edit --global grub-boot-success.timer
```
creates/edits the file `/etc/systemd/user/grub-boot-success.timer.d/override.conf`.

Now I get that we're not doing exactly the same thing here, because we're creating drop-ins in `/usr/lib` and not `/etc`.  There might be a use case for creating user drop-ins for specific users in the future though, maybe as a user customization or for `anaconda` or something.  If we do want that, I'm not sure what the stage options would look like right now, because we would need an option that can take the values `system`, `global`, or `user` but with a username as well.

@thozza Dragging you in for stage option design.  WDYT?


[1] https://www.freedesktop.org/software/systemd/man/systemctl.html#edit%20UNIT%E2%80%A6

Reply:
I'm slightly inclining towards the enum, since it would keep our options open. As for the schema, it would probably end up looking something like https://github.com/osbuild/osbuild/blob/main/inputs/org.osbuild.files if we need to allow additional properties for various enum values.

Reply:
For now let's make it an enum with just `system` and `global` as options
```json
"type": {
  "enum": ["system", "global"]
}
```

and then if we want to add user it can become

```json
"oneOf": [
  {
    "required": ["type"],
    "additionalProperties": false,
    "properties": {
      "type": {
        "enum": ["system", "global"]
      }
    }
  },
  {
    "required": ["type", "user"],
    "additionalProperties": false,
    "properties": {
      "type": {
        "enum": ["user"]
      },
      "user": {
        "type": "string"
      }
    }
  }
]
```

Reply:
that looks good.  I would rename the top-level key from `systemd-options` to `unit-type`.  Unless someone has a better idea.